cellprofiler/modules/yeast_cell_segmentation.py

```diff
@@ -0,0 +1,892 @@
+        cellstar.parameters["segmentation"]["maxOverlap"] = self.maximal_cell_overlap.value
+        if self.advanced_cell_filtering.value:
+            def calculate_area_multiplier(area):
+                return 4.0 * area / self.average_cell_diameter.value ** 2 / math.pi
+
+            def calculate_size_multiplier(area):
+                return calculate_area_multiplier(area) ** 0.5
+
+            areas_range = self.min_cell_area.value, self.max_cell_area.value
+            cellstar.parameters["segmentation"]["minArea"] = max(cellstar.parameters["segmentation"]["minArea"], calculate_area_multiplier(areas_range[0]))
+            cellstar.parameters["segmentation"]["maxArea"] = calculate_area_multiplier(areas_range[1])
+            # to some extent change length of rays
+            #cellstar.parameters["segmentation"]["stars"]["maxSize"] = max(cellstar.parameters["segmentation"]["stars"]["maxSize"], min(2.5, calculate_size_multiplier(areas_range[1])))
+
+        success = cellstar.decode_auto_params(self.autoadapted_params.value)
+        if not success:  # if current value is invalid overwrite it with current settings
+            self.autoadapted_params.value = cellstar.encode_auto_params()
+        return cellstar
+
+    #
+    # Segmentation of the image into yeast cells.
+    # Returns: yeast cells, yeast cells qualities, background
+    #
+    def segmentation(self, normalized_image, background_pixels):
+        cellstar = self.prepare_cell_star_object(self.segmentation_precision.value)
+
+        if self.input_image_file_name is not None:
+            dedicated_image_folder = pj(pref.get_default_output_directory(), self.input_image_file_name)
+            if dedicated_image_folder is not None:
+                cellstar.debug_output_image_path = dedicated_image_folder
+
+        cellstar.set_frame(normalized_image)
+        cellstar.set_background(background_pixels)
+        segmented_image, snakes = cellstar.run_segmentation()
+
+        objects = cellprofiler.objects.Objects()
+        objects.segmented = segmented_image
+        objects.unedited_segmented = segmented_image
+        objects.small_removed_segmented = np.zeros(normalized_image.shape)
+        # objects.parent_image = normalized_image has to be cellprofiler image
+        
+        #if self.current_workspace.frame is not None:
+        self.current_workspace.display_data.segmentation_pixels = objects.segmented
+
+        return objects, np.array([-s.rank for s in snakes]), cellstar.images.background
+
+    def ground_truth_editor( self ):
+        '''Display a UI for GT editing'''
+        from cellprofiler.gui.editobjectsdlg import EditObjectsDialog
+        from wx import OK
+        import wx
+        #title = "%s #%d, image cycle #%d: " % (self.module_name,
+        #                                     self.module_num,
+        #                                     image_set_number)
+        
+        ### opening file dialog
```

Comment:
wx.FileDialog is not destroyed. Maybe use a "with" block:

    with wx.FileDialog(...) as dlg: